cluster-autoscaler/utils/drain/drain_test.go

```diff
@@ -624,7 +642,73 @@ func TestDrain(t *testing.T) {
 		},
 	}
 
-	for _, test := range tests {
+	// run all tests for scaleDownNodesWithCustomControllerPods=false
+	// TODO(vadasambar): remove this when we get rid of scaleDownNodesWithCustomControllerPods
+	for _, test := range append(tests, testOpts{
```

Comment:
I am doing `append` in the for loop declaration itself. This might not be the best approach. I didn't want to create 2 variables called `testsWithScaleDownNodesWithCustomControllerPodsDisabled` and `testsWithScaleDownNodesWithCustomControllerPodsEnabled`. Any better suggestions are welcome.

Reply:
What is wrong about being explicit? I'd something along the lines of:
1. Make custom controller pods flag a part of the test case struct.
2. Define a list of shared test cases.
3. Define a list of flag-disabled test cases (copy the shared ones and add the extra one)
4. Define a list of flag-enabled test cases (copy the shared ones, flip the flag on all and add the extra one)
5. Define the body of the test once and run it for a combined list (flag-enabled + flag-disabled).

WDYT?

Reply:
Alternatively, just copy all test cases and have a single list in the first place :)

Reply:
Thank you for the suggestion. I like the idea of putting the flag on the test struct and combining all tests into one. Let me try it out.

Reply:
Updated the code https://github.com/kubernetes/autoscaler/pull/5507/files#diff-cd1a98b8de97414dad2b66a3cf3229b1572a169086d9b7141a266f8ddb597f68